extension/source/currencies/defaultCurrency.ts

```diff
@@ -0,0 +1,21 @@
+import * as io from 'io-ts';
+
+import assertType from '../cells/assertType';
+
+export default async function defaultCurrency(): Promise<string> {
+  try {
+    const ipBasedCurrencyJson = await fetch(
+      'http://ip-api.com/json/?fields=currency',
+    ).then((res) => res.json());
```

Comment:
I don't think some people are going to like it if we're letting a 3rd party service look them up by IP location, even though this could happen if we decided to log ips on a aggregator or the same on a chain RPC node.

I would recommend doing one of the following instead to bootstrap someone's initial currency:
- Use [navigator.language](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/language) and map that to a currency.
- Ask for permission to use `http://ip-api.com/json/?fields=currency'` to detect someone's currency
- Use [navigator.geolocation.getCurrentPosition](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition) to allow the user to decide if they want to share their position. Then translate those coordinates to a country. Note if you do this you will also need to update our extension manifest with the ['geolocation' permission](https://developer.chrome.com/docs/extensions/mv3/declare_permissions/)

Reply:
> letting a 3rd party service look them up by IP location

I think there might be a misunderstanding here. Every website and external service you use gets to see your IP address and can use that IP to look up your location. Services don't ask permission for your IP because it's a pre-requisite of using the internet.

Users wishing to hide their local IP need to use a VPN or similar. I don't really see the benefit of avoiding ip-api.com.

If this isn't new information to you, perhaps we could chat about it further?

Reply:
Updated to just use USD as discussed.